front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/PipelineController.java

```diff
@@ -275,6 +286,20 @@ private PipelineTemplateDAO getTemplateDAO() {
                 "Pipeline Templates are not supported with your current storage backend"));
   }
 
+  private void checkForStalePipeline(Pipeline pipeline, GenericValidationErrors errors) {
+    Pipeline existingPipeline = pipelineDAO.findById(pipeline.getId());
```

Comment:
I feel like there is a case here where there isn't an existing pipeline (via the save entrypoint), which should probably be legit from a stale checking perspective. I also suspect the submitted payload in that case would not have a last modified (unless that is magically set somewhere along the way with a default value).

Reply:
My intent was to use this for the deck save pipeline dialog, in which case it would only request stale checking for existing pipelines.  But yeah, might as well make it more robust.  I'm thinking if the submitted payload has an ID (i.e., it's an update op, not an insert) and stale check is requested, we can bomb out in surprising scenarios like the existing pipeline not found.

> unless that is magically set

I think `lastModified` is set when the entity is actually persisted, so yeah good call.